app/screens/login/login.js

```diff
@@ -73,18 +74,25 @@ export default class Login extends PureComponent {
         setMfaPreflightDone(false);
     }
 
-    componentWillReceiveProps(nextProps) {
-        if (this.props.loginRequest.status === RequestStatus.STARTED && nextProps.loginRequest.status === RequestStatus.SUCCESS) {
-            this.props.actions.handleSuccessfulLogin().then(this.goToChannel);
-        } else if (this.props.loginRequest.status !== nextProps.loginRequest.status && nextProps.loginRequest.status !== RequestStatus.STARTED) {
-            this.setState({isLoading: false});
+    componentDidUpdate(nextProps) {
+        const {loginRequest: {status}, actions: {handleSuccessfulLogin}} = this.props;
+
+        if (status === RequestStatus.STARTED && nextProps.loginRequest.status === RequestStatus.SUCCESS) {
+            handleSuccessfulLogin().then(this.goToChannel);
         }
     }
 
     componentWillUnmount() {
         Dimensions.removeEventListener('change', this.orientationDidChange);
     }
 
+    static getDerivedStateFromProps(nextProps, prevState) {
```

Comment:
You're never updating `this.state.loginRequest` with this function, so it will always have whatever initial value is passed when this component is mounted. `getDerivedStateFromProps` should return an object containing the `loginRequest` whenever that changes.

Reply:
I would totally get rid of the `loginRequest` prop and this lifecycle method and handle it based on the login action returned value.

check how it was done for the realm migration here https://github.com/mattermost/mattermost-mobile/blob/multi-server/app/screens/login/login.js